cmd/skopeo/inspect.go

```diff
@@ -142,6 +143,29 @@ func (opts *inspectOptions) run(args []string, stdout io.Writer) (retErr error)
 		return nil
 	}
 
+	var signers []string
+	sig, err := img.Signatures(ctx)
+	if err != nil {
+		return fmt.Errorf("Error reading image signatures: %v", err)
+	}
+	if len(sig) > 0 {
+		mech, err := signature.NewGPGSigningMechanism()
+		if err != nil {
+			return fmt.Errorf("Error preparing local GPG: %v", err)
+		}
+		for _, s := range sig {
+			_, id, err := mech.UntrustedSignatureContents(s)
+			if err == nil {
+				_, kID, err := mech.Verify(s)
```

Comment:
(This verifies against `GNUPGHOME`, not the keys in `policy.json`.)

More importantly, it does not verify the signature matches the rest of the image at all! An attacker could attach any valid signature to any other image.